nidmviewerfsl/lib/statformat.py

```diff
@@ -218,3 +218,53 @@ def contrastVec(data, v_min, v_max):
 
     # Return the image
     return('data:image/jpg;base64,' + encodedIm.decode())
+
+
+# This function takes in an excursion set name and a graph and generates
+# an FSL-like name for the HTML output display for the excursionset.
+#
+# e.g. ExcursionSet_F001.nii.gz -> cluster_zfstat1_std.html
+def getClusFileName(g, excName):
+
+    # For SPM data we can't work out the filename we want from just
+    # the contrast name.
+    if runQuery(g, 'askSPM', 'Ask'):
+
+        # For SPM data we must look for the statistic map to
+        # assert which statistic is associated to a contrast.
+        statisticMap = runQuery(g, 'selectStatMap', 'Select',
+                                {'EXC_NAME': excName})[0]
+
+        # If it's T stat string is '', if it's F statstring
+        # is 'f'
+        if statisticMap[0] == 'T':
+            statString = ''
+        else:
+            statString = statisticMap[0].lower()
+
+        return('cluster_z' + statString + 'stat1_std.html')
```

Comment:
This is only for the SPM use case. The reason this is currently hard coded is because the missing link for inference objects issue I reported [here](https://github.com/incf-nidash/nidmresults-spm/issues/51) means that if there were more than one excursion set map in an SPM-generated NIDM-Results pack, I would not be able to work out which statistic maps were associated with it. This means for the moment this section of the code assumes there is only one excursion set map for each SPM-generated NIDM-Results pack.

Reply:
Okay. I understand! In that case, the clearest way to deal with this problem would be to add the missing relation in the document when reading it and then just use the normal code here. This means adding a condition when reading a NIDM doc that will check for exporter type and version and if it's SPM with the problematic version then applies the fix. Similar to what is done [here](https://github.com/cmaumet/nidmresults/blob/aa89327e5a2cc82959746b1c99baa9a23277779b/nidmresults/graph.py#L59-L118).

Reply:
Hi @cmaumet ,

Rather embarrassingly the error I was talking about above did not exist and trying to implement the fix you suggested led to me to realize I had made a mistake! I have removed the github issue on the nidmresults-spmhtml repository explaining I was mistaken and I have updated the code so that this query now works. The index that originally started this thread however is still coded to one, as there should only be one statistic map per excursion set.